impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultModelXmlFactory.java

```diff
@@ -121,7 +121,7 @@ public void write(XmlWriterRequest<Model> request) throws XmlWriterException {
         Writer writer = request.getWriter();
         Function<Object, String> inputLocationFormatter = request.getInputLocationFormatter();
         if (writer == null && outputStream == null && path == null) {
-            throw new IllegalArgumentException("writer, outputStream or path must be non null");
+            throw new IllegalArgumentException("writer, output stream, or path must be non-null");
```

Comment:
That's wrong and unrelated to this PR

Reply:
```suggestion
            throw new IllegalArgumentException("writer, output stream, or path must be non null");
```
you mean the `hyphen -` ?

Reply:
No, `outputStream` is the parameter name, it's not `output stream`.

Reply:
yes true. Making it generic would take impl details away. What if tomorrow foo resources is new input? 
This is kind of fluff. We need some valid input thats all.
```suggestion
            throw new IllegalArgumentException("No valid input source given.");
```

Reply:
But that's not helpful to the user.  What does 'valid' means ?

Reply:
Yes, we could tell, as it is.
```suggestion
            throw new IllegalArgumentException("Given input source is null.");
```

Reply:
Considering `OOP` leveraging domain logic its kind of clear from the context now as the code should speak for itself:

`ReadRequest/WriteRequest` 
<img width="1865" alt="image" src="https://github.com/user-attachments/assets/0af239f8-122b-46cb-84f6-8bd7762050ad" />

Reply:
We should focus on communicating the actual business case rather than exposing low-level implementation details, which are likely to change.

At this stage, it’s more a matter of preference, since the code now clearly delineates each concern. Code and logging are 2 different worlds.